packages/clarity-visualize/src/layout.ts

```diff
@@ -41,7 +41,18 @@ export function markup(event: Layout.DomEvent, iframe: HTMLIFrameElement): void
     let doc = iframe.contentDocument;
     for (let node of data) {
         let parent = element(node.parent);
-        let next = element(node.next);
+        let pivot = element(node.previous);
+        let insert = insertAfter;
+
+        // For backward compatibility. Until v0.4.4 we used the next element to determine right ordering within DOM
+        // Starting with v0.4.5, we moved away from the next element and instead starting sending previous element
+        // This change fixes several edge cases where positioning of DOM could get inconsistent.
+        // In future, we can get rid of following code, but keeping it for now to ensure backward compatibility.
+        if ("next" in node) {
```

Comment:
Question: Would it be better to explicitly check the version of the data you are receiving rather than checking for a field? This is a judgment call but I have a slight preference for explicitly including the version in the logic here so that there is no ambiguity and no potential future problem.

Reply:
I have a slight preference for feature detection over version checking but in this case even if we wanted to enforce version check, we will need to change the interface for visualization and that will turn into a bigger change. In future I do want to update the interface such that we can't call replay without passing in the version. Today it's possible to send in events without ever sending the envelope. I think it's not great but want to address that shortcoming as part of a different change.